api/src/main/java/org/pmiops/workbench/monitoring/StackdriverStatsExporterService.java

```diff
@@ -34,23 +49,68 @@ public StackdriverStatsExporterService(Provider<WorkbenchConfig> workbenchConfig
   public void createAndRegister() {
     if (!initialized) {
       try {
-        final StackdriverStatsConfiguration configuration =
-            StackdriverStatsConfiguration.builder()
-                .setMetricNamePrefix(buildMetricNamePrefix())
-                .setProjectId(workbenchConfigProvider.get().server.projectId)
-                .build();
+        final StackdriverStatsConfiguration configuration = makeStackdriverStatsConfiguration();
         StackdriverStatsExporter.createAndRegister(configuration);
+        logger.info(
+            String.format(
+                "Configured StackDriver exports with configuration:\n%s",
+                configuration.toString()));
         initialized = true;
       } catch (IOException e) {
         logger.log(Level.WARNING, "Failed to initialize global StackdriverStatsExporter.", e);
       }
     }
   }
 
+  @VisibleForTesting
+  public StackdriverStatsConfiguration makeStackdriverStatsConfiguration() {
+    return StackdriverStatsConfiguration.builder()
+        .setMetricNamePrefix(buildMetricNamePrefix())
+        .setProjectId(getProjectId())
+        .setMonitoredResource(makeMonitoredResource())
+        .build();
+  }
+
+  private MonitoredResource makeMonitoredResource() {
+    final MonitoredResource.Builder resultBuilder =
+        MonitoredResource.newBuilder()
+            .setType(MONITORED_RESOURCE_TYPE)
+            .putLabels(PROJECT_ID_LABEL, getProjectId())
+            .putLabels(LOCATION_LABEL, getLocation())
+            .putLabels(NAMESPACE_LABEL, getEnvironmentShortName())
+            .putLabels(NODE_ID_LABEL, getNodeId());
+
+    return resultBuilder.build();
+  }
+
+  private String getProjectId() {
+    return workbenchConfigProvider.get().server.projectId;
+  }
+
+  private String getLocation() {
+    return workbenchConfigProvider.get().server.appEngineLocationId;
```

Comment:
Is there a plan to have this return the actual location in the future?  Please add a comment.

Reply:
Plan would be to have this set in the config at deploy time based on a `gcloud` command. It's app-wide.

Can't add comments to the JSON file tho.

Reply:
OK